datameta/api/ui/view.py

```diff
@@ -224,6 +224,7 @@ def post(request: Request):
                 submission_datetime   = mdata_set.submission.date.isoformat(),
                 submission_label      = mdata_set.submission.label,
                 service_executions    = service_executions,
+                replaced_by           = get_identifier(mdata_set.replaced_via_event.new_metadataset) if mdata_set.replaced_via_event else None
```

Comment:
This will fire an additional two queries for every `mdata_set`, first when accessing the `replaced_via_event` attribute, then when accessing the `new_metadataset` attribute. Since we know we will be accessing those fields, we can instruct SQLAlchemy not to use lazy loading, but to join the corresponding relations already when we query the `mdata_sets`. Please add the corresponding joins in L177.